implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OtlpMeterRegistryTest.java

```diff
@@ -67,9 +66,10 @@ abstract class OtlpMeterRegistryTest {
     @BeforeEach
     void setUp() {
         this.clock = new MockClock();
+        OtlpConfig config = otlpConfig();
         this.mockHttpSender = mock(HttpSender.class);
-        this.registry = new OtlpMeterRegistry(otlpConfig(), this.clock,
-                new NamedThreadFactory("otlp-metrics-publisher"), this.mockHttpSender);
+        OtlpMetricsSender metricsSender = new OtlpHttpMetricsSender(mockHttpSender, config);
+        this.registry = OtlpMeterRegistry.builder(config).clock(clock).metricsSender(metricsSender).build();
```

Comment:
One thing I worry about that is demonstrated here is how it's possible to pass a different instance of `OtlpConfig` to the `OtlpHttpMetricsSender` and the `OtlpMeterRegistry` (or Builder). It would be better if such a misconfiguration weren't possible, but we can think on how to improve this in the next milestone rather than block merging this for early feedback, I think.